synchronization/src/main/java/de/cyface/synchronization/WiFiSurveyor.java

```diff
@@ -336,22 +336,7 @@ public void makeAccountSyncable(@NonNull final Account account, boolean enabled)
         ContentResolver.setSyncAutomatically(account, authority, false);
         setSyncEnabled(account, enabled);
 
-        validateAccountFlags(account, authority);
-    }
-
-    /**
-     * Makes sure the account flags used by {@link #isConnected()} are valid.
-     * <p>
-     * See {@link #makeAccountSyncable(Account, boolean)} for details.
-     *
-     * @param account The {@code Account} to be checked.
-     */
-    public static void validateAccountFlags(@NonNull final Account account, @NonNull final String authority) {
-        final boolean periodicSyncRegistered = ContentResolver.getPeriodicSyncs(account, authority).size() > 0;
-        final boolean autoSyncEnabled = ContentResolver.getSyncAutomatically(account, authority);
-        Validate.isTrue(periodicSyncRegistered == autoSyncEnabled,
-                "Both, periodicSync and autoSync, must be in the same state but are: " + periodicSyncRegistered
-                        + " and " + autoSyncEnabled + ", in this order");
+            // Do not use validateAccountFlags in production code as periodicSync flags are set async
```

Comment:
This is the important change in the production code:
- As the periodicSync flag is changed asynchronously we must not check that the change was successful immediately. This throws an exception in the STAD app. For the same reason I moved the validation method to the test flavour to make sure we don't accidentally  use this again in production.
- there is a WifiSurveyorTest which validates that the periodicSync flag is set

Reply:
As the public validateAccountFlags method was not used by others and is not used anymore I did not change the 1st digit code version for this file.